include/wil/registry_basic.h

```diff
@@ -1040,39 +1026,31 @@ namespace reg_view_details
         typename err_policy::result set_value(_In_opt_ PCWSTR subkey, _In_opt_ PCWSTR value_name, const R& value) const
         {
             const auto error = ::RegSetKeyValueW(
-                ::wil::registry::details::get_key(m_key),
+                ::wil::reg::details::get_key(m_key),
                 subkey,
                 value_name,
-                ::wil::registry::reg_view_details::set_value_type(value),
-                static_cast<BYTE*>(::wil::registry::reg_view_details::get_buffer(value)),
-                ::wil::registry::reg_view_details::get_buffer_size(value));
+                ::wil::reg::reg_view_details::set_value_type(value),
+                static_cast<BYTE*>(::wil::reg::reg_view_details::get_buffer(value)),
+                ::wil::reg::reg_view_details::get_buffer_size(value));
             return err_policy::HResult(HRESULT_FROM_WIN32(error));
         }
 
-        // TODO: just call set_value with nullptr subkey?
         template <typename R>
         typename err_policy::result set_value(_In_opt_ PCWSTR value_name, const R& value) const
         {
-            const auto error = RegSetValueExW(
-                ::wil::registry::details::get_key(m_key),
-                value_name,
-                0,
-                ::wil::registry::reg_view_details::set_value_type(value),
-                static_cast<BYTE*>(::wil::registry::reg_view_details::get_buffer(value)),
-                ::wil::registry::reg_view_details::get_buffer_size(value));
-            return err_policy::HResult(HRESULT_FROM_WIN32(error));
+            return set_value(nullptr, value_name, value);
         }
 
-#ifdef WIL_ENABLE_EXCEPTIONS
+#if defined(__WIL_WINREG_STL)

```

Comment:
Does `__WIL_WINREG_STL` imply `WIL_ENABLE_EXCEPTIONS`, or do we still need to check both in places like this?